Travis failure:
```
bitcoin_cli.py failed, Duration: 1 s
stdout:
2017-11-17 13:47:24.232000 TestFramework (INFO): Initializing test directory /tmp/bitcoin_test_runner_20171117_134443/bitcoin_cli_303
2017-11-17 13:47:24.492000 TestFramework (INFO): Compare responses from gewalletinfo RPC and `bitcoin-cli getwalletinfo`
2017-11-17 13:47:24.503000 TestFramework (INFO): Compare responses from getblockchaininfo RPC and `bitcoin-cli getblockchaininfo`
2017-11-17 13:47:24.509000 TestFramework (INFO): Test -stdinrpcpass option
2017-11-17 13:47:24.768000 TestFramework (INFO): Test -stdin and -stdinrpcpass
2017-11-17 13:47:25.030000 TestFramework (INFO): Compare responses from `bitcoin-cli -getinfo` and the RPCs data is retrieved from.
2017-11-17 13:47:25.041000 TestFramework (ERROR): Unexpected exception caught during testing
Traceback (most recent call last):
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 120, in main
    self.run_test()
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-i686-pc-linux-gnu/test/functional/bitcoin_cli.py", line 39, in run_test
    cli_get_info = self.nodes[0].cli('-getinfo').help()
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_node.py", line 211, in dispatcher
    return self.send_cli(command, *args, **kwargs)
  File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-i686-pc-linux-gnu/test/functional/test_framework/test_node.py", line 229, in send_cli
    raise subprocess.CalledProcessError(returncode, self.binary, output=cli_stderr)
subprocess.CalledProcessError: Command '/home/travis/build/bitcoin/bitcoin/build/bitcoin-i686-pc-linux-gnu/src/bitcoin-cli' returned non-zero exit status 1
2017-11-17 13:47:25.042000 TestFramework (INFO): Stopping nodes
```

--- a/src/bitcoin-cli.cpp
+++ b/src/bitcoin-cli.cpp
@@ -213,6 +213,9 @@ public:
     /** Create a simulated `getinfo` request. */
     UniValue PrepareRequest(const std::string& method, const std::vector<std::string>& args) override
     {
+        if (!args.empty()) {
+            throw std::runtime_error("-getinfo takes no arguments");
+        }
         UniValue result(UniValue::VARR);
         result.push_back(JSONRPCRequestObj("getnetworkinfo", NullUniValue, ID_NETWORKINFO));
         result.push_back(JSONRPCRequestObj("getblockchaininfo", NullUniValue, ID_BLOCKCHAININFO));

--- a/test/functional/bitcoin_cli.py
+++ b/test/functional/bitcoin_cli.py
@@ -35,8 +35,11 @@ class TestBitcoinCli(BitcoinTestFramework):
         assert_equal(["foo", "bar"], self.nodes[0].cli('-rpcuser=%s' % user, '-stdin', '-stdinrpcpass', input=password + "\nfoo\nbar").echo())
         assert_raises_process_error(1, "incorrect rpcuser or rpcpassword", self.nodes[0].cli('-rpcuser=%s' % user, '-stdin', '-stdinrpcpass', input="foo").echo)
 
+        self.log.info("Make sure that -getinfo with arguments fails")
+        assert_raises_process_error(1, "-getinfo takes no arguments", self.nodes[0].cli('-getinfo').help)
+
         self.log.info("Compare responses from `bitcoin-cli -getinfo` and the RPCs data is retrieved from.")
-        cli_get_info = self.nodes[0].cli('-getinfo').help()
+        cli_get_info = self.nodes[0].cli().send_cli('-getinfo')
         wallet_info = self.nodes[0].getwalletinfo()
         network_info = self.nodes[0].getnetworkinfo()
         blockchain_info = self.nodes[0].getblockchaininfo()

